src/jobs/keys-update/keys-update.service.ts

```diff
@@ -147,13 +147,24 @@ export class KeysUpdateService {
           // update staking module information
           await this.srModulesStorage.upsert(contractModule, currNonce);
 
-          // now updating decision should be here moduleInstance.update
-          // TODO: operators list also the same ?
-          if (moduleInStorage && prevNonce === currNonce) {
-            // nothing changed, don't need to update
-            this.logger.log(
-              `Nonce was not changed for staking module ${moduleInStorage.id}. Don't need to update keys and operators in database`,
-            );
+          this.logger.log(`Nonce previous value: ${prevNonce}, nonce current value: ${currNonce}`);
+
+          if (prevNonce === currNonce) {
+            // case when prevELMeta is undefined but prevNonce === currNonce looks like invalid
+            if (
+              prevElMeta &&
+              prevElMeta.blockNumber + 1 <= currElMeta.number &&
```

Comment:
Maybe for readability just check `prevElMeta.blockNumber < currElMeta.number &&` ?